### PR TITLE
Inject Serving images

### PR DESCRIPTION
**Pull request by skonto**
- Part of the work defined in SIG Release
- Replaces Serving images with the ones built on CI via env vars
- Gets manifests from the mid stream repo
- Ingress will come as a next step

### CODE REVIEW

**Comment by skonto** [comment]
Serving tests faiure:
AutoscaleSustainingTest 
TestAutoscaleSustaining/aggregation-weighted-exponential

**Comment by skonto** [comment]
```
error: unable to recognize "STDIN": Get "https://api.ci-ocp-4-10-amd64-aws-us-east-1-dsbnd.hive.aws.ci.openshift.org:6443/api?timeout=32s": dial tcp: lookup api.ci-ocp-4-10-amd64-aws-us-east-1-dsbnd.hive.aws.ci.openshift.org on 172.30.0.10:53: no such host
```

File: olm-catalog/serverless-operator/project.yaml
Comment (pierDipi) [inline olm-catalog/serverless-operator/project.yaml:31]: nit: maybe group this with `dependencies.serving` as put them closer to each other?

File: openshift-knative-operator/hack/update-manifests.sh
Comment (skonto) [inline openshift-knative-operator/hack/update-manifests.sh:45]: After we merge the PR at the openshift/serving repo we can then change back to `https://raw.githubusercontent.com/openshift/knative-serving/${branch}/openshift/release/artifacts/$index-$file`

Comment (pierDipi) [inline openshift-knative-operator/hack/update-manifests.sh:45]: nit: if we align the serving repo first we can point to the proper midstream branch instead of yours, meaning we can merge your branch in serving first, and then we can just use the final one here.

This would simplify the review of the deleted patches making sure that what we're removing here is added to the serving repo.

File: openshift-knative-operator/hack/update-manifests.sh
Comment (pierDipi) [inline openshift-knative-operator/hack/update-manifests.sh:119]:
```suggestion
# Instead openshift/knative-serving uses its own generated ci manifests and sets KNATIVE_SERVING_TEST_MANIFESTS_DIR.
```

File: openshift-knative-operator/hack/002-openshift-serving-role.patch
Comment (pierDipi) [inline openshift-knative-operator/hack/002-openshift-serving-role.patch:4]: Great! Another step to the removal of versioned directories!

File: openshift-knative-operator/hack/001-serving-namespace-deletion.patch
Comment (pierDipi) [inline openshift-knative-operator/hack/001-serving-namespace-deletion.patch:1]: I don't see this patch in https://github.com/skonto/serving/tree/image_injection_1.3/openshift/patches, maybe it's planned for later but I'd keep track of these removed patches.

**Review by pierDipi — COMMENTED** [review]
Great, another little step!

File: openshift-knative-operator/hack/update-manifests.sh
Comment (matzew) [reply openshift-knative-operator/hack/update-manifests.sh:45]: +1 we did same

Comment (matzew) [reply openshift-knative-operator/hack/update-manifests.sh:45]: or that way - yeah

File: openshift-knative-operator/hack/001-serving-namespace-deletion.patch
Comment (pierDipi) [reply openshift-knative-operator/hack/001-serving-namespace-deletion.patch:1]: Ok, thanks!

One note on patches: 
- https://github.com/skonto/serving/blob/image_injection_1.3/openshift/release/manifest-patches/001-serving-namespace-deletion.patch is based on a generated bundles, is there any reason for not doing something like this https://github.com/openshift-knative/eventing-kafka-broker/blob/2f16a913ea82e6b36d0ccc7e4eb5f5deea7c0f1b/openshift/release/generate-release.sh#L10-L11, that would remove the need to change `"serving.knative.dev/release: "v1.3.0"` to `serving.knative.dev/release: "v1.4.0"` in the next release.

Similar comment on other patches since basing them on the origin of the final manifests would benefit from future changes.

Comment (pierDipi) [reply openshift-knative-operator/hack/001-serving-namespace-deletion.patch:1]: But that's not really relevant for this PR, just a thought

Comment (skonto) [reply openshift-knative-operator/hack/001-serving-namespace-deletion.patch:1]: Yeah we could do that I will take a note and do it when we merge the patches. As a side note  https://github.com/skonto/serving/tree/image_injection_1.3/openshift/patches, contains also code patches in [https://github.com/skonto/serving/blob/image_injection_1.3/openshift/release/manifest-patches](https://github.com/skonto/serving/blob/image_injection_1.3/openshift/release/manifest-patches/001-serving-namespace-deletion.patch) it is only manfest patches as they are used in Serverless Operator.

**Comment by skonto** [comment]
@matzew @pierDipi ready, gentle ping.

File: openshift-knative-operator/hack/update-manifests.sh
Comment (pierDipi) [inline openshift-knative-operator/hack/update-manifests.sh:121]: I guess TODO is done?
```suggestion
# repo.
```

File: openshift-knative-operator/hack/update-manifests.sh
Comment (pierDipi) [reply openshift-knative-operator/hack/update-manifests.sh:119]: Missing nit

**Review by pierDipi — APPROVED** [review]
/hold
2 nits, they can be fixed later tho

/lgtm
/approve

**Pull request merged**
